FreeRTOSIPConfig.h

```diff
@@ -42,7 +42,34 @@ connected. */
 //extern void printf( const char *pcFormatString, ... );
 #include <stdio.h>
 
-#define ipconfigBUFFER_PADDING 14
+/* Space left at the beginning of a network buffer storage area to store a
+ * pointer back to the network buffer.  Should be a multiple of uintptr_t to ensure uintptr_t byte
+ * alignment is maintained on architectures that require it.
+ *
+ * In order to get a 32-bit alignment of network packets, an offset of 2 bytes
+ * would be desirable, as defined by ipconfigPACKET_FILLER_SIZE.  So the malloc'd
+ * buffer will have the following contents:
+ *  uintptr_t pointer;   // uintptr_t-aligned
+ *  uchar_8 filler[6];
+ *  << ETH-header >>    // half-word-aligned
+ *  uchar_8 dest[6];    // start of pucEthernetBuffer
+ *  uchar_8 dest[6];
+ *  uchar16_t type;
+ *  << IP-header >>     // word-aligned
+ *  uint8_t ucVersionHeaderLength;
+ *  etc
+ */
+#ifdef __CHERI_PURE_CAPABILITY__
+    #if __riscv_xlen == 32
+        #define ipconfigBUFFER_PADDING    14
+    #else
+        #define ipconfigBUFFER_PADDING    22
+    #endif
+#else
+    #if __riscv_xlen == 64
+        #define ipconfigBUFFER_PADDING    14
+    #endif
```

Comment:
If `__riscv_xlen` is 32 and `__CHERI_PURE_CAPABILITY_` is undefined, the buffer padding is left undefined. Is that what we want?

Reply:
Yes, it’s defined by default by FreeRTOS-TCP which assumes 32-bit builds. These defines here are added because anything other than 32-bit is considered custom

Reply:
In FreeRTOS_IP.h:

```
/* Space left at the beginning of a network buffer storage area to store a
 * pointer back to the network buffer.  Should be a multiple of 8 to ensure 8 byte
 * alignment is maintained on architectures that require it.
 *
 * In order to get a 32-bit alignment of network packets, an offset of 2 bytes
 * would be desirable, as defined by ipconfigPACKET_FILLER_SIZE.  So the malloc'd
 * buffer will have the following contents:
 *  uint32_t pointer;   // word-aligned
 *  uchar_8 filler[6];
 *  << ETH-header >>    // half-word-aligned
 *  uchar_8 dest[6];    // start of pucEthernetBuffer
 *  uchar_8 dest[6];
 *  uchar16_t type;
 *  << IP-header >>     // word-aligned
 *  uint8_t ucVersionHeaderLength;
 *  etc
 */

    #if ( ipconfigBUFFER_PADDING != 0 )
        #define ipBUFFER_PADDING    ipconfigBUFFER_PADDING
    #else
        #define ipBUFFER_PADDING    ( 8U + ipconfigPACKET_FILLER_SIZE )
    #endif
```